sql/handler.cc

```diff
@@ -2136,6 +2138,14 @@ int ha_rollback_to_savepoint(THD *thd, SAVEPOINT *sv)
     status_var_increment(thd->status_var.ha_savepoint_rollback_count);
     trans->no_2pc|= ht->prepare == 0;
   }
+
+  /*
+    Write tables map once more for trigger as it will be wiped out in
+    Write_rows_log_event::write_row() on slave.
+  */
+  if (thd->in_sub_stmt == SUB_STMT_TRIGGER)
+    error= write_locked_table_maps(thd, true);
```

Comment:
One minor issue re. error flag, but I'm not sure whether it's worth fixing. ha_rollback_to_savepoint sets it disregarding any previous value in the flag for each ht->*_rollback call. The failure of these calls might as well be an assert failure only, as the comments "cannot happen" say, because the main way they would fail in InnoDB is if the savepoint name is not found, which is validated earlier. (Not sure about TokuDB).

But with this new call the rollback may also fail if we have non-trx tables in the binlog event group and we get a disk full error while writing the ROLLBACK event to binlog, which is quite unlikely but not a "cannot happen" error.

If you think it's worth addressing, s/error= /error|= /g, otherwise merge it or let me know I should merge it.